specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/DatabaseRecommendedActions.json

```diff
@@ -0,0 +1,585 @@
+      "required": [
+        "currentValue"
+      ],
+      "type": "object",
+      "properties": {
+        "currentValue": {
+          "description": "Current state the recommended action is in. Some commonly used states are: Active      -> recommended action is active and no action has been taken yet. Pending     -> recommended action is approved for and is awaiting execution. Executing   -> recommended action is being applied on the user database. Verifying   -> recommended action was applied and is being verified of its usefulness by the system. Success     -> recommended action was applied and improvement found during verification. Pending Revert  -> verification found little or no improvement so recommended action is queued for revert or user has manually reverted. Reverting   -> changes made while applying recommended action are being reverted on the user database. Reverted    -> successfully reverted the changes made by recommended action on user database. Ignored     -> user explicitly ignored/discarded the recommended action. ",
+          "enum": [
+            "Active",
+            "Pending",
+            "Executing",
+            "Verifying",
+            "PendingRevert",
+            "RevertCancelled",
+            "Reverting",
+            "Reverted",
+            "Ignored",
+            "Expired",
+            "Monitoring",
+            "Resolved",
+            "Success",
+            "Error"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "RecommendedActionCurrentState",
+            "modelAsString": true
+          }
+        },
+        "actionInitiatedBy": {
+          "description": "Gets who initiated the execution of this recommended action. Possible Value are: User    -> When user explicity notified system to apply the recommended action. System  -> When auto-execute status of this advisor was set to 'Enabled', in which case the system applied it.",
+          "enum": [
+            "User",
+            "System"
+          ],
+          "type": "string",
+          "readOnly": true,
+          "x-ms-enum": {
+            "name": "RecommendedActionInitiatedBy",
+            "modelAsString": false
+          }
+        },
+        "lastModified": {
+          "format": "date-time",
+          "description": "Gets the time when the state was last modified",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    }
+  },
+  "parameters": {
+    "DatabaseNameParameter": {
+      "name": "databaseName",
+      "in": "path",
+      "description": "The name of the database.",
+      "required": true,
+      "type": "string",
+      "x-ms-parameter-location": "method"
+    },
+    "ServerNameParameter": {
+      "name": "serverName",
+      "in": "path",
+      "description": "The name of the server.",
+      "required": true,
+      "type": "string",
+      "x-ms-parameter-location": "method"
+    }
+  },
```

Comment:
patterns are added in DatabaseOperations.json's /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Sql/servers/{serverName}/databases/{databaseName}/operations .

Why the inconsistency ?

Reply:
Yes, why doesn't the regex pattern allow the example property value to be 'Default-SQL-SouthEastAsia'?

>String does not match pattern ^[\p{L}\p{N}().-]*[\p{L}\p{N}()-]$:

Reply:
It actually seems to be the ResourceGroupName regex which is the problem?

>        "name": "resourceGroupName",
            "in": "path",
            "description": "The name of the resource group that contains the resource.",
            "required": true,
            "type": "string",
            "maxLength": 90,
            "minLength": 1,
            "pattern": "^[\\p{L}\\p{N}_().-]*[\\p{L}\\p{N}_()-]$"

Reply:
I think it would be better if you were using a reference ($ref) to common types definition of ResourceGroupParameter 

https://github.com/Azure/azure-rest-api-specs/edit/main/specification/common-types/resource-management/v5/types.json

Reply:
i.e. the regex isn't correct vs the spec:

https://learn.microsoft.com/en-us/azure/azure-resource-manager/management/resource-name-rules#microsoftresources